src/Common/EventRateMeter.h

```diff
@@ -27,6 +27,14 @@ class EventRateMeter
     /// NOTE: Adding events into distant past (further than `period`) must be avoided.
     void add(double now, double count)
     {
+        // Remove data for initial heating stage that can present at the beginning of a query.
+        // Otherwise it leads to wrong gradual increase of average value, turning algorithm into not very reactive.
+        if (count != 0.0 && ++data_points < 5)
```

Comment:
can we provide any reasoning why 5 is chosen here?

Reply:
no reason, pure heuristic :) actually it comes from data I saw on simple queries: 3 or 4 initial data points are for some reason show less cpu consumption. I guess there should be a reason for that "heat up", maybe something in server can be fixed, but it can be hard to find, so this simple heuristic looks okay.